### DISCUSSION TITLE
zh/guide/advanced/offline-download

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#141] [Announcements]
# zh/guide/advanced/offline-download

请仔细阅读以下提示 有两款离线下载功能软件 两款用法都一样，提前设置好软件配置(具体配置看下面说明)， 然后回到前端界面， ==找到你想把离线文件上传到哪个文件夹，你就手动进入到该文件夹== 然后再右下角找到离线下载按钮，自行选择使用 Aria2下载还是qBittorrent下载(＞v 3.11.0) 1.Aria2 点击查看使用说明 (../../c...

https://alist.nn.ci/zh/guide/advanced/offline-download.html

**Comment by fly-pigawa** [comment]
大大，Aria2的链接显示404打不开（

**Reply by anwen-anyi** [reply]
输错了一个符合~就是这个介绍
https://alist.nn.ci/zh/config/other.html

**Comment by kdxcxs** [comment]
`admin:adminadmin@localhost:8080` 应该就是默认配置，我当时稍微搜了一下说是这个就这样写了，另外这个配置也可以在 alist 后台修改了，具体格式可以去搜一下 url 的格式，这里放一张图吧：

![image.png](https://s2.loli.net/2023/02/20/56ETKqMJV2dsZRB.png)

如果在 qBittorrent 设置了 alist 所在的 ip 免验证的话可以直接省略 `@` 前面的用户名和密码。

另外，因为用到了 `v2.8.3` 的 api，所以 qBittorrent 的版本最低应该是 `4.4.0beta2` 或者 `4.3.8` 正式版。

**Reply by anwen-anyi** [reply]
嗯 看到啦~不过视频已经做完了 哈哈 就不加了...可以在文档提一下吧简单的

**Comment by kermonds** [comment]
您好，出现这种情况如何处理呢？
![网页显示](https://i.imgtg.com/2023/02/21/s1ZGC.png)
![实际上服务器已经收到请求了](https://i.imgtg.com/2023/02/21/s1KrN.png)

**Reply by kermonds** [reply]
网页端显示unknown error occurred downloading qbittorrent
但是服务器端已经get到请求了，但不会自动执行上传

**Reply by anwen-anyi** [reply]
现在用的哪个版本Alist v3.11.0吗？这个版本现在有BUG~ 你试试action的测试版先哪些都是修复了的

**Reply by kermonds** [reply]
好的，我试试

**Reply by kermonds** [reply]
大佬这测试版在哪里下？我翻不到········

**Reply by kdxcxs** [reply]
https://github.com/alist-org/alist/suites/11104300911/artifacts/565581352

**Reply by kermonds** [reply]
好了，用了https://github.com/alist-org/alist/actions/runs/4232378334
![s1jCP.png](https://i.imgtg.com/2023/02/21/s1jCP.png)

**Reply by wanglongfei8899** [reply]
大佬能不能在前端 加一个配置 qBittorrent的窗口啊

**Reply by kdxcxs** [reply]
已经有了 https://github.com/alist-org/alist-web/commit/bcb174a5e26a4c861e970c9433b4f68301452ebe ，下一个正式版里面应该就会看到

**Comment by wanglongfei8899** [comment]
大佬能不能在前端 加一个配置 qBittorrent的窗口啊

**Reply by anwen-anyi** [reply]
在前端加做什么？ 又不像Aria2那样 可以让用户本地下载调用

**Reply by kdxcxs** [reply]
测试版里面已经有了，等下一个版本吧

**Reply by anwen-anyi** [reply]
加了什么？没看到诶 在前端界面~

**Reply by kdxcxs** [reply]
![image.png](https://s2.loli.net/2023/02/22/sUQjxV1TKuPLlrX.png)

**Reply by anwen-anyi** [reply]
好吧后台,这个我倒是看到了
我以为在首页呢..我看他说前端（首页）  哈哈

**Reply by wanglongfei8899** [reply]
自己改db文件打不开 不想用默认的账号和密码

**Comment by lonelypers** [comment]
alist添加用户和密码后提示：
Post "http://localhost:8080/api/v2/auth/login": dial tcp [**1]:8080: connect: connection refused

大佬们指点一下啊

**Reply by lonelypers** [reply]
8080端口已放行了

**Reply by Hypergear** [reply]
用docker的话加一行--network host

**Comment by Wdtu** [comment]
qb只能下载磁力么，怎么下载种子呢？

**Reply by kdxcxs** [reply]
是的，目前只能下载磁链，可以试试 [torrent to magnet](https://nutbread.github.io/t2m/) 这种工具把种子转成磁链下载

**Comment by 15109827573** [comment]
大大，我使用的docker的alist，已经映射/opt/alist/data/temp/aria2和/opt/alist/data/temp/qbittorrent两个路径，使用离线下载时出现以下情况

![This is an image](https://s1.locimg.com/2023/03/03/e7cf6562e7cf0.png)

经过排查，离线下载时，都能成功调用aria2和qbittorrent，但软件会在/opt/alist/data/temp/aria2和/opt/alist/data/temp/qbittorrent（这两个路径没有就会自己创建，是实际路径，不是映射路径）下载文件夹和文件。上传时却是找的映射路径，也就是在/mnt/sda1/aria2和/mnt/sda1/qbittorrent（映射后的实际路径）里找文件，所以就会出现上图提示的“找不到文件夹或文件”。
映射目录是根据网上教程，修改了hostconfig.json的Binds数组和config.v2.json的MountPoints，在docker管理界面直接点击路径显示映射成功，在/mnt/sda1/aria2和/mnt/sda1/qbittorrent存储文件，在docker管理界面直接点击路径也能看到该文件。到这里就不知道问题出在哪里了，请大大帮忙看看

![This is an image](https://s1.locimg.com/2023/03/03/30b862fab6efe.png)
容器管理界面能显示映射路径

![This is an image](https://s1.locimg.com/2023/03/03/3e407ed266398.png)
这是alist调用aria2离线下载后，aria2在/opt/alist/data/temp/aria2自行创建相应文件开始下载

![This is an image](https://s1.locimg.com/2023/03/03/a59f7c140d748.png)
这是实际地址/mnt/sda1/aria2，却没有文件

**Reply by 15109827573** [reply]
重新docker安装，使用docker run -d --restart=always -v /mnt/mmcblk2p4/alist:/opt/alist/data -v /mnt/sda1/qbittorrent:/opt/alist/data/temp/qbittorrent -v /mnt/sda1/aria2:/opt/alist/data/temp/aria2 -p 5244:5244 -e PUID=0 -e PGID=0 -e UMASK=022 --name="alist" xhofe/alist-aria2:v3.12.2 。离线下载时问题依旧，问题一模一样。

**Reply by 15109827573** [reply]
问题解决了，真的很给力。确实需要aria2和alist在同一容器里，我一开始使用的是openwrt的插件版aria2版本，在设置那里也提示成功了，但没想到就这步引发问题。话不多说，解决办法是用容器里的Aria2。alist网络是桥接方式，内部地址是172.31.0.2。Aria2设置地址是http://172.31.0.2:6800/jsonrpc。完美通过测试。插件版aria2出现的问题，估计是用户权限的问题。

**Reply by 15109827573** [reply]
刚刚测试了下qbittorrent，跟之前的一开始的问题一样。强烈建议将qbittorrent也集合到容器里。

**Reply by Hypergear** [reply]
怎么在docker里面装Aria啊😂

**Reply by Hypergear** [reply]
写错了怎么在docker里面装qbittorrent啊😂

**Reply by Jerry12228** [reply]
> 写错了怎么在docker里面装qbittorrent啊😂

参考
https://github.com/qbittorrent/docker-qbittorrent-nox#readme

**Comment by lclrc** [comment]
你好, aria2离线下载到阿里云盘有点问题诶. 我用的是`xhofe/alist-aria2:main`的amd64镜像. 试了两次都是下载到临时文件夹的内容和大小都是对的, 往网盘上传出现问题, 监控服务器流量基本没变化, alist网页管理那的上传进度也一直卡在最开始的位置, 最后都是经过一个多小时后显示已完成但上传的大小有问题. 第一次是传了100M, 第二次是传了60M.
PS: 测试下载的是个Centos7minimal的torrent, 里面几个KB级别的小文件能上传成功, 973M的iso镜像上传有问题.
另外测试了两次离线下载到mega盘都是正常的.
PPS: 以后会出同时内置aria2和qb的docker吗?
麻烦大大了!

**Comment by Jerry12228** [comment]
有大佬在Centos7编译qb 4.5.2 nox 成功的吗，我这报错装不上呜呜呜

**Reply by Jerry12228** [reply]
···
FAILED: src/base/CMakeFiles/qbt_base.dir/qbt_base_autogen/mocs_compilation.cpp.o 
/opt/rh/devtoolset-10/root/usr/bin/g++ -DBOOST_ASIO_ENABLE_CANCELIO -DBOOST_ASIO_NO_DEPRECATED -DDISABLE_GUI -DOPENSSL_NO_SSL2 -DQBT_USES_LIBTORRENT2 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050f02 -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_DEBUG_OUTPUT -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_SQL_LIB -DQT_STRICT_ITERATORS -DQT_USE_QSTRINGBUILDER -DQT_XML_LIB -DTORRENT_LINKING_SHARED -DTORRENT_SSL_PEERS -DTORRENT_USE_LIBCRYPTO -DTORRENT_USE_OPENSSL -I/root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/include -I/root/qBittorrent-release-4.5.2/src -isystem /usr/local/openssl/include -isystem /opt/Qt5.15.8/include -isystem /opt/Qt5.15.8/include/QtCore -isystem /opt/Qt5.15.8/./mkspecs/linux-g++ -isystem /opt/Qt5.15.8/include/QtNetwork -isystem /opt/Qt5.15.8/include/QtSql -isystem /opt/Qt5.15.8/include/QtXml -O3 -DNDEBUG -fexceptions -fPIC -Wall -Wextra -Wcast-qual -Wcast-align -Winvalid-pch -Woverloaded-virtual -Wold-style-cast -Wnon-virtual-dtor -pedantic -pedantic-errors -Wstrict-null-sentinel -std=c++17 -MD -MT src/base/CMakeFiles/qbt_base.dir/qbt_base_autogen/mocs_compilation.cpp.o -MF src/base/CMakeFiles/qbt_base.dir/qbt_base_autogen/mocs_compilation.cpp.o.d -o src/base/CMakeFiles/qbt_base.dir/qbt_base_autogen/mocs_compilation.cpp.o -c /root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/mocs_compilation.cpp
In file included from /root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/N5CYPY6UQL/moc_server.cpp:10,
                 from /root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/mocs_compilation.cpp:17:
/root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/N5CYPY6UQL/../../../../../src/base/http/server.h:66:17: error: field ‘m_key’ has incomplete type ‘QSslKey’
   66 |         QSslKey m_key;
      |                 ^~~~~
In file included from /opt/Qt5.15.8/include/QtNetwork/QSslCertificate:1,
                 from /root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/N5CYPY6UQL/../../../../../src/base/http/server.h:34,
                 from /root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/N5CYPY6UQL/moc_server.cpp:10,
                 from /root/qBittorrent-release-4.5.2/build/src/base/qbt_base_autogen/mocs_compilation.cpp:17:
/opt/Qt5.15.8/include/QtNetwork/qsslcertificate.h:63:7: note: forward declaration of ‘class QSslKey’
   63 | class QSslKey;
      |       ^~~~~~~
[23/175] Building CXX object src/base/CMakeFiles/qbt_base.dir/torrentfileswatcher.cpp.o
ninja: build stopped: subcommand failed.
···

**Comment by wanglei52076** [comment]
Alist使用群晖套件版，aria2和qt使用docker版本，配置都正确，离线下载不成功，看日志应该是权限问题
回复怎么上传图片的 这个还不会玩 ，直接日志打出来吧：
[util.cc:1948] errNum=13 errorCode=18 Failed to make the directory /volume1/@appdata/alist3/temp/aria2/8ff36762-ff7f-4127-9de3-5745cde18f98, cause: Permission denied

find: '/volume1/@appdata/alist3/temp/aria2/8ff36762-ff7f-4127-9de3-5745cde18f98":No such file or directory

后续能修正这个问题吗？还是说通过群晖授予套件最高执行权限？

**Reply by wanglei52076** [reply]
解决了 直接PUID PGID 赋予root权限 root的puid和pgid都是0 以后这种烦人的权限问题就这么解决算了 不过要提前开启root 这个百度教程很多！

**Comment by Hypergear** [comment]
如果qbit和alist都是docker部署的, 应该咋整啊

**Reply by Hypergear** [reply]
解决了docker启动alist的时候要把网络模式改成host(--network host)

**Comment by huanity** [comment]
分享一个VPS上配置aria2离线下载的过程：
aria2位置：主机
alist位置：docker容器
alist后台配置：http://主机ip:6800/jsonrpc + 密钥
docker映射：-v /opt/alist/data/temp/aria2:/opt/alist/data/temp/aria2

解释：在alist面板调用aria2离线下载请求时，会将下载目录指定为/opt/alist/data/temp/aria2，而后的transfer也将会在这个目录中寻找文件，在docker中，下载完成后会在alist容器内部寻找该目录（/opt/alist/data/temp/aria2），而此时主机aria2会将文件下载于主机目录/opt/alist/data/temp/aria2，因此需要进行目录映射才能让alist找到文件

同样的，假如aria2在docker中，文件下载位置将会在aria2容器内部/opt/alist/data/temp/aria2中，因此在启动aria2容器时需要将一个主机目录（假设为a）映射到Aira2容器目录/opt/alist/data/temp/aria2，然后将a再映射到alist容器中的/opt/alist/data/temp/aria2，相当于a目录起到容器间目录桥接的作用

**Reply by wanglei52076** [reply]
尝试docker -v  /opt/alist/data/temp/aria2:/opt/alist/data/temp/aria2 还是提示错误，离线不成功！还是等作者更新修复吧，容器二合一版本没有问题！

**Comment by Jerry12228** [comment]
温馨提示：使用docker部署qb的同学不要忘记将alist的temp目录挂载到qb容器
eg. 如下是我的
-v /data/alist/temp:/data/alist/temp

**Comment by AvalonO98** [comment]
使用qbit下载完成后上传时报错unexpected EOF，是什么原因呢

**Comment by Jerry12228** [comment]
qbit离线下载失败，添加任务，下载完元数据后任务状态变成错误，alist显示错误码为error:error
alist位于主机，qbit位于docker，已开特权模式，已将alist的temp目录挂载到qbit容器
目前分析为qbit无权写挂载的目录，因为在主机中将alist的temp目录权限设置为777即可恢复正常，但每次重启alist时，该目录权限会重置为755

怎么办，谢谢
有关qbit的进程信息（宿主机查看）
![image](https://user-images.githubusercontent.com/86408775/230584200-1b1774c4-02be-49d8-8b2c-96041b74cc3f.png)

**Reply by Purange** [reply]
同样的问题

**Reply by Jerry12228** [reply]
> 同样的问题

我解决了
写了个教程（有帮助请给个三连谢谢）
【Centos中使用Alist和qBittorrent（Docker）实现离线下载-哔哩哔哩】 https://b23.tv/S7KjUCd

**Reply by Purange** [reply]
太感谢了。。解决了困扰我好久的问题！

**Comment by buzidjy** [comment]
debian11安装的alist。阿里网盘open使用qBittorrent离线下载，下载成功后上传秒成功，网盘里也有文件但是大小完全不对，一个电影18G但是网盘里的文件只有20M。有人知道这是怎么回事么？

**Reply by Purange** [reply]
我最近也遇到这个问题了，之前几百兆的动画都没这个问题。感觉可能跟alist的上传机制有关，就算你正常上传一个过大的文件也经常失败

**Reply by learningdog1** [reply]
我成功下载并上传过几个G的文件，但是当磁力出现多个文件时，alist会上传小文件，最后大文件会忽略。

**Comment by stevenwongh** [comment]
大佬请问一下，aria2下载显示成功，文件呢？temp目录是空的。容器的/opt/alist/data映射到宿主机的目录了，那么容器里其子目录/opt/alist/data/temp/aria2应该也是保存到数据机上的啊，为什么下载完就清空了，下载的文件在哪里呢？求解答

**Reply by stevenwongh** [reply]
用的是docker预装aria2的版本，但是下载后文件就没了，显示成功

**Comment by cjw414522569** [comment]
用openwrt的兄弟们注意！！！，用openwrt的alist插件和aria2插件的，注意aria2用户名要选择root，不然离线下载会报错无法创建文件（权限不够）

**Reply by cjw414522569** [reply]
补充一点，aria2的下载目录要设置在alist缓存目录

**Comment by mingtin** [comment]
下载成功，转移却失败了。出现如下提示，请问各位大佬，这是怎么回事？

%!w(<nil>); CreateTempFile failed, incoming stream actual size= %!s(int64=1019032711), expect = %!s(int64=903150711)

**Comment by cryboy007** [comment]
离线下载转存天翼网盘报错：
code: SliceMd5DoesNotMatch ,msg: sessionKey=f8e64571-49ae-4c0d-bdd3-e6eb95a97a99, sliceMd5 do not match! requestId=484c3898-68bd-4aca-8c78-cdfaca4cfe48

**Comment by CC342** [comment]
alist和qb都是docker版本，下载问题，也显示上传成功。但是，单个文件最大就是20M，不管上传的文件本身有几个G,上传成功就是20M最大，求解，

**Reply by learningdog1** [reply]
可能是Qbittorrent做种时间原因，目前我测试一个磁力中有多个文件时，只要上传完一个小文件，就会立马删除任务，导致大文件还没上传完，就发现已经没有文件了。

**Reply by learningdog1** [reply]
修改做种时间也无效  搞蒙了 也不知道问题在哪了

**Comment by glassywater** [comment]
如果想要qBittorrent无限时间做种的话该怎么设置呢

**Comment by akronblim** [comment]
如果在alist中使用非管理员账号，那么qbittorrent无法添加离线下载，提示Permission denied，无论我更改qbt的权限还是alist的权限都不行，全部都是root权限启动的，仍然提示如此。在debian 12测试过，在centos 8测试过，qbt使用nox enhanced edition

**Comment by 15109827573** [comment]
您的来信我已收到

**Comment by xxcyou3** [comment]
![疑难杂症](https://img2.y4cc.cc/file/abfa5d7e4fe91e6f3044b.jpg)
离线下载后文件名字是这个怎么解决，兄弟们

**Comment by 15109827573** [comment]
您的来信我已收到

**Comment by renjiu13** [comment]


**Reply by hamuzoo** [reply]
文件名字太长有没有解决办法？
failed to download 640_0DdiTCuVurakv-FJR, error: failed to download 271105c181df933d, error: Failed to make the directory /opt/alist/data/temp/aria2/00aef78d-18a7-4695-9a88-8604857aa019/[2024.03.20] ラブライブ！蓮ノ空女学院スクールアイドルクラブ るりのとゆかいなつづりたち&かほめぐ♡じぇらーと&蓮ノ休日 シャッフルユニットスプリットシングル「Colorfulness／ハッピー至上主義！／Pleasure Feather」[MP3 320K], cause: File name too long

**Comment by hamuzoo** [comment]
文件名字太长有没有解决办法？
failed to download 640_0DdiTCuVurakv-FJR, error: failed to download 271105c181df933d, error: Failed to make the directory /opt/alist/data/temp/aria2/00aef78d-18a7-4695-9a88-8604857aa019/[2024.03.20] ラブライブ！蓮ノ空女学院スクールアイドルクラブ るりのとゆかいなつづりたち&かほめぐ♡じぇらーと&蓮ノ休日 シャッフルユニットスプリットシングル「Colorfulness／ハッピー至上主義！／Pleasure Feather」[MP3 320K], cause: File name too long

**Comment by 15109827573** [comment]
您的来信我已收到

**Comment by toneoesa** [comment]
踩了个大坑啊😭，alist 启动的时候会把 `/opt/alist/data/temp/qbittorrent` 清空😭（是这样吗？还是其他问题？），怎么没有地方警告一下呢😭。
因为我的 qbittorrent 也是 docker 部署的，为了方便，理所当然地把我原本的下载目录也挂载到上面的目录了，就像这样：
```
    volumes:
      - /path/to/my/qbittorrent:/downloads
      - /path/to/my/qbittorrent:/opt/alist/data/temp/qbittorrent
```
然后悲剧的事情就发生了，qbittorrent 下载目录全空了😭😭😭。

提醒大家不要像我这样部署啊。

**Reply by toneoesa** [reply]
发现了一个相似问题的 issue/disussion 在[这里](https://github.com/alist-org/alist/discussions/4097#discussion-5061351)。

**Reply by toneoesa** [reply]
在[配置文件-temp_dir](https://alist.nn.ci/zh/config/configuration.html#temp-dir)处有警告，不过离这里也太远了

**Reply by SeanChengN** [reply]
佛了，没了几百G文件，看媒体库才发现文件没了

**Comment by 15109827573** [comment]
您的来信我已收到

**Comment by GavinYe123** [comment]
操作时间2024年7月30日。大佬您好，我在参照着您的文档进行配置时发现了一个问题：alist在控制docker版本的qbittorrent时，在容器内的下载路径为“/opt/alist/data/temp/qBittorrent”，而参照文档中描述到的关于docker版qbittorrent的映射目录内外均为“/opt/alist/data/temp/qbittorrent”，参照文档进行配置的docker版本qbittorrent无法正常工作。不知道我是否仅仅是个例，如是普遍性问题应该修改一下文档

**Reply by GavinYe123** [reply]
修正：映射的目录应该内外都为“/opt/alist/data/temp/qBittorrent”，字母B均为大写

**Comment by SalicylicAcid** [comment]
离线下载 - 每行一条链接 下面的“从不删除”是什么意思？是在下载并保存到目标位置之后在本地留存一个副本吗？文件路径又在哪？

**Reply by xiayeroot** [reply]
你在前台某个文件夹页面提交离线下载时，提交的任务下载完成会自动上传到那个文件夹，但是下载时文件会保存到配置的下载路径，如果选择 从不删除，下载完成的文件上传到文件夹中之后，下载路径的文件还会存在，设置这个的目的我猜测可能是某些下载工具比如qBittorrent的部分下载服务器会有分享需求，比如你下载1G的文件会要求你上传1G的流量，否则之后再去下载会限速，相当于让你再分享文件出去，这样留存文件达到分享需求后就可以在下载工具中再手动删除

**Comment by LunaticXJ** [comment]
大佬，115 Cloud的离线下载能否开放到API中去？

**Comment by xiayeroot** [comment]
您好，我发现alist提交了任务之后，比如离线下载，如果中途重启了alist或者主机，提交的任务会消失掉，但是下载工具中途重启还可以继续下载，aria2我看了是把下载信息存放到一个.session文件中，重启后会读取出来继续下载，请问alist能否实现类似功能，万分感谢

**Comment by YCZ01111** [comment]
要是alist可以支持实现115网盘的rss订阅就好了

**Comment by stilldo** [comment]
现在不支持115 Cloud 离线下载了吗   右下角没有选项
